src/libLookup/Lookup.cpp

```diff
@@ -3351,22 +3351,32 @@ bool Lookup::CommitTxBlocks(const vector<TxBlock>& txBlocks) {
           m_rejoinInProgress = false;
           m_rejoinNetworkAttempts = 0;  // reset rejoining attempts
           cv_setRejoinRecovery.notify_all();
-          if (m_lookupServer) {
-            if (m_lookupServer->StartListening()) {
-              LOG_GENERAL(INFO, "API Server started to listen again");
-            } else {
-              LOG_GENERAL(WARNING, "API Server couldn't start");
-            }
-          }
           m_isFirstLoop = true;
+          auto startJsonRpc = [this]() mutable -> void {
+            LOG_GENERAL(INFO, "In startJsonRpc func");
+            std::unique_lock<std::mutex> cv_lk(m_mutexJsonRpcPortOpen);
+            if (m_mediator.m_lookup->cv_jsonRpcPortOpen.wait_for(
```

Comment:
should consider checking some flag before waiting? Otherwise, here it will always wait for 180 sec before starting listening, if `listening stops `finished before `listening start` is called.
Flag should be set/reset before notify.